src/Microsoft.Identity.Client/ApiConfig/AbstractAcquireTokenParameterBuilder.cs

```diff
@@ -340,7 +351,7 @@ internal void ValidateAndCalculateApiId()
             Validate();
             CommonParameters.ApiId = CalculateApiEventId();
             CommonParameters.ApiTelemId = ApiTelemetryId;
-            CommonParameters.TelemetryCorrelationId = Guid.NewGuid();
+            CommonParameters.TelemetryCorrelationId = (CommonParameters.UserProvidedTelemetryCorrelationId != null) ? CommonParameters.UserProvidedTelemetryCorrelationId : new Guid();
```

Comment:
you should use Guid.NewGuid() instead of "new Guid()".  I don't believe the Guid constructor actually calls CoCreateGuid()